File: test/src/ctypes_test.c
// Copyright (C) 2008-2010 by Yung-Yu Chen.  See LICENSE.txt for terms of
// usage.
#include "ctypes.h"
void ctypes_test(int *par_i, double *par_d, int *n_i, int *par_arr_i,
		int *n_d2, int *n_d1, double *par_arr_d) {
	int it, jt;
	for (it=0; it<n_i[0]; it++) {
		par_arr_i[it] += par_i[0];
	};
	for (it=0; it<n_d1[0]; it++) {
		for (jt=0; jt<n_d2[0]; jt++) {
			par_arr_d[it*n_d2[0]+jt] += par_d[0];
		};
	};
	for (it=0; it<n_d2[0]; it++) {
		par_arr_d[it] -= par_d[0];
	};
	return;
};
// vim: set ts=4 et:

File: obselete/testsrc/ctypes_test.f90
subroutine ctypes_test(par_i, par_d, n_i, par_arr_i, n_d1, n_d2, par_arr_d)
implicit none
! arguments.
integer*4, intent(in) :: par_i
real*8, intent(in) :: par_d
integer*4, intent(in) :: n_i
integer*4, intent(inout) :: par_arr_i(n_i)
integer*4, intent(in) :: n_d1, n_d2
real*8, intent(inout) :: par_arr_d(n_d1, n_d2)

par_arr_i(:) = par_arr_i(:) + par_i
par_arr_d(:,:) = par_arr_d(:,:) + par_d
par_arr_d(:,1) = par_arr_d(:,1) - par_d
end subroutine ctypes_test
